### PR TITLE
add diskspace healthcheck checker

### PR DESCRIPTION
**Pull request by josledp**
Solves issue #6

### CODE REVIEW

File: checkers/diskspace.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/etherlabsio/healthcheck"
+)
+
+type diskspace struct {
+	filesystem string
```
Comment (karthikmuralidharan) [inline checkers/diskspace.go]: Could this be renamed to `dir` ?

File: checkers/diskspace.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	if err != nil {
+		return fmt.Errorf("Error looking for fs stats: %v", err)
+	}
+
+	total := fs.Blocks * uint64(fs.Bsize)
+	free := fs.Bfree * uint64(fs.Bsize)
+	used := total - free
+
+	if 100*used/total > ds.threshold {
+		return fmt.Errorf("Space used on %s greater than threshold %d%%(%d%%)", ds.filesystem, ds.threshold, 100*used/total)
```
Comment (karthikmuralidharan) [inline checkers/diskspace.go]: Could the first letter in the message be in lower-case ?

File: checkers/diskspace.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+type diskspace struct {
+	filesystem string
+	threshold  uint64
+	statfs     func(string, *syscall.Statfs_t) error
+}
+
+//Check test if the filesystem disk usage is above threshold
+func (ds *diskspace) Check() error {
+	if _, err := os.Stat(ds.filesystem); err != nil {
+		return fmt.Errorf("Filesystem not found: %v", err)
```
Comment (karthikmuralidharan) [inline checkers/diskspace.go]: Could the first letter in the message be in lower-case ?

File: checkers/diskspace.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	fs := syscall.Statfs_t{}
+	err := ds.statfs(ds.filesystem, &fs)
+	if err != nil {
+		return fmt.Errorf("Error looking for fs stats: %v", err)
```
Comment (karthikmuralidharan) [inline checkers/diskspace.go]: Should we specify the directory name in the error as well ?

**Comment by karthikmuralidharan** [comment]
Sorry for the delayed response @josledp and thanks for the contribution !!

Overall the implementation looks good to me. I have left some minor comments. Please check it out.

**Comment by josledp** [comment]
I've updated the PR. Let me now if you have any other concern :)

**Review by karthikmuralidharan — APPROVED** [review]
👍

**Pull request merged**